services/business/checkers/route_rules/precedence_checker.go

```diff
@@ -0,0 +1,54 @@
+package route_rules
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/models"
+)
+
+type PrecedenceChecker struct{ kubernetes.IstioObject }
+
+func (route PrecedenceChecker) Check() ([]*models.IstioCheck, bool) {
+	valid := true
+	validations := make([]*models.IstioCheck, 0)
+
+	if route.GetSpec()["precedence"] == nil {
+		return validations, valid
+	}
+
+	precedence, err := castPrecedence(route.GetSpec()["precedence"])
+	if err != nil {
+		valid = false
+		validation := models.BuildCheck("Precedence must be a number",
+			"error", "spec/precedence/"+route.GetSpec()["precedence"].(string))
```

Comment:
Ok, so now I see, the Reference of the error is not on absolute # of a line but in the element that is offending the rule.
This is good, as then we would be able to search it easily.